File: topic2-performance-parallelism/src/time_write.py
import time
import sys
import os

def main():
    # Check if user provided input
    if len(sys.argv) != 2:
        print("Usage: python3 time_write.py <number_of_lines>")
        sys.exit(1)

    # Number of lines to write
    n = int(sys.argv[1])

    # Create data folder if it doesn't exist
    os.makedirs("data", exist_ok=True)

    filename = "data/sample_data.txt"

    # Start timing
    start_time = time.time()

    # Write to file
    with open(filename, "w") as f:
        for i in range(n):
            f.write(f"This is line {i}\n")

    # End timing
    end_time = time.time()

    # Calculate runtime
    run_time = end_time - start_time

    # Output results
    print(f"Wrote {n} lines to {filename}")
    print(f"Runtime: {run_time:.6f} seconds")

if __name__ == "__main__":
    main()

File: topic2-performance-parallelism/src/time_write.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main(int argc, char *argv[]) {
    if (argc != 2) {
        printf("Usage: ./time_write <number_of_lines>\n");
        return 1;
    }

    int n = atoi(argv[1]);

    FILE *file = fopen("data/sample_data.txt", "w");
    if (file == NULL) {
        printf("Error opening file.\n");
        return 1;
    }

    clock_t start_time = clock();

    for (int i = 0; i < n; i++) {
        fprintf(file, "This is line %d\n", i);
    }

    fclose(file);

    clock_t end_time = clock();
    double run_time = (double)(end_time - start_time) / CLOCKS_PER_SEC;

    printf("Wrote %d lines to data/sample_data.txt\n", n);
    printf("Runtime: %.6f seconds\n", run_time);

    return 0;
}
